test/integration/framework/framework.go

```diff
@@ -177,3 +178,30 @@ func waitForNamespaceDeletion(client kubeclientset.Interface, namespace string)
 	}
 	return nil
 }
+
+func (f *TestFramework) GenerateCustomImagePullSecret(nsName string) error {
+	if TestContext.DockerUserName != "" && TestContext.DockerUserPsd != "" {
+		client := f.GetKubeClient(fmt.Sprintf("%s-create-namespace", f.BaseName))
+		_, err := client.CoreV1().Secrets(nsName).Create(context.TODO(), dockerConfigSecret(CustomImagePullSecretName, nsName, TestContext.DockerUserName, TestContext.DockerUserPsd), metav1.CreateOptions{})
+		if err != nil {
+			Logf("Failed to create secret in the namespace <%s>: %v", nsName, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func dockerConfigSecret(secName, nsName, dockerUserName, dockerUserPassword string) *corev1.Secret {
```

Comment:
You should also have cases where .travis.yaml does not specify $DOCKER_USERNAME and $DOCKER_PASSWORD

Reply:
If .travis.yaml does not specify $DOCKER_USERNAME and $DOCKER_PASSWORD, the code will not create the secret. the pod may still have the `imagePullsecret` set, but it will not bother or stop the pod from pullinig image with the default secret as `imagePullsecret` is always optional for a pod. you could specify one or more  `imagePullsecret`  in a pod even some of them may not even exist.